src/app/dataValidator/dataValidator.js

```diff
@@ -10,16 +10,54 @@ const yamlSchema = fs.readFileSync('./././data/schema.yaml', 'utf8');
 const { components } = yaml.load(yamlSchema);
 const { schemas } = components;
 
-const validateNode = (currentSchemaNode, dataNode) => {
-  validateType(currentSchemaNode.type, dataNode);
+const validateNode = (currentSchemaNode, dataNode, schemaName) => {
+  validateType(currentSchemaNode.type, dataNode, schemaName);
 
   if (currentSchemaNode.enum) {
-    validateEnum(currentSchemaNode.enum, dataNode);
+    validateEnum(currentSchemaNode.enum, dataNode, schemaName);
   }
 
   if (currentSchemaNode.required) {
-    validateRequired(currentSchemaNode.required, dataNode);
+    validateRequired(currentSchemaNode.required, dataNode, schemaName);
   }
+
+  if (currentSchemaNode.properties) {
+    validateProperties(currentSchemaNode, dataNode, schemaName); // eslint-disable-line no-use-before-define
+  }
+};
+
+const isRequiredProperty = (property, schema) => {
+  if (schema.required) {
+    if (schema.required.includes(property)) {
+      return true;
+    }
+  }
+  return false;
+};
+
+/*
+  Due to the recursive nature of properties being able to be nested inside of properties
+  both validateProperties() and validateNode() need to be contained within a single file
+  otherwise a ReferenceError is throw due to the other method not yet being defined
+*/
+const validateProperties = (currentSchemaNode, dataNode, schemaName) => {
```

Comment:
`ValidateBlock` should be in its own separate file. 
Likewise, the `validateProperties` and `validateNode` functions should be in a separate file. I understand that due to their recursive nature they need to be in the same file, but they don't need to be in this particular file here.

Reply:
This was addressed during the file restructuring